src/BibleWell.Aquifer.Api/AquiferApiService.cs

```diff
@@ -54,4 +55,106 @@ private class ResourceContentResponse
         public required string Name { get; set; }
         public required IReadOnlyList<string> Content { get; set; }
     }
+
+    // --- push notifications ---
+    private static CreateDeviceInstallationRequest MapToCreateDeviceInstallationRequest(DeviceInstallation source)
+    {
+        return new CreateDeviceInstallationRequest()
+        {
+            InstallationId = source.InstallationId,
+            Platform = source.Platform,
+            PushChannel = source.PushChannel,
+            Tags = source.Tags,
+        };
+    }
+    
+    public async Task RegisterDeviceAsync(DeviceInstallation deviceInstallation)
+    {
+        try
+        {
+            await _aquiferClient.PushNotifications
+                .DeviceInstallation
+                .PostAsync(MapToCreateDeviceInstallationRequest(deviceInstallation))
+                .ConfigureAwait(false);
+        }
+        catch (ErrorResponse ex)
+        {
+            _logger.LogError(ex, "RegisterDeviceAsync failed.");
+            throw;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Error creating or updating device installation on Aquifer Well API.");
+            throw;
+        }
+    }
+
+    public async Task DeRegisterDeviceAsync(string deviceId)
+    {
+        try
+        {
+            await _aquiferClient.PushNotifications.DeviceInstallation[deviceId].DeleteAsync().ConfigureAwait(false);
+        }
+        catch (ErrorResponse ex)
+        {
+            _logger.LogError(ex, "DeRegisterDeviceAsync failed.");
+            throw;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Error: device installation deletion on Aquifer Well API.");
+            throw;
+        }
+    }
+    
+    // --- demo code for requesting notifications ---
+    public async Task RequestActionAAsync()
```

Comment:
It shouldn't be hard to do these demo tests with Postman and then the client will never need this code.  We should probably hide these routes from the generated client, too.